docs/tp/04.md

```diff
@@ -0,0 +1,60 @@
+---
+sidebar_position: 4
+description: Utilisation des descripterus de fichers et dup2
+---
+
+# TP 04 - Fichiers et descripteurs de fichiers
+
+## Resources
+ 1. [OpenOptions](https://doc.rust-lang.org/std/fs/struct.OpenOptions.html)
+ 2. [Stdin](https://doc.rust-lang.org/std/io/fn.stdin.html)
+ 3. [Stdout](https://doc.rust-lang.org/std/io/fn.stdout.html)
+ 4. [dup](https://docs.rs/nix/0.23.1/nix/unistd/fn.dup.html)
+ 5. [dup2](https://docs.rs/nix/0.23.1/nix/unistd/fn.dup2.html)
+
+## Linux Standard Streams
+
+Sous Linux, `stdin` est le flux d'entrée standard. Cela accepte le texte comme entrée. La sortie texte de la commande au shell est transmise via le flux `stdout` (sortie standard). Les messages d'erreur de la commande sont envoyés via le flux `stderr` (erreur standard).
```

Comment:
Both Windows and Linux use this, as `stdin` is part of the C standar library. Fd 0, 1 and 2 are Lunux specific.

Reply:
Done